state/storage.go

```diff
@@ -797,7 +806,7 @@ func readStorageConstraints(st *State, key string) (map[string]StorageConstraint
 	var doc storageConstraintsDoc
 	err := coll.FindId(key).One(&doc)
 	if err == mgo.ErrNotFound {
-		return nil, nil
+		return nil, errors.NotFoundf("storage constraints for %q", key)
```

Comment:
I am not sure why you'd want to return an error if no constraints were found... Is it not ok not to have any constraints?